src/transformStateWithClones.js

```diff
@@ -7,7 +7,55 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const arr = [];
+  const clone = { ...state };
+
+  for (let i = 0; i < actions.length; i++) {
+    const actionObj = actions[i];
+
+    switch (actionObj.type) {
+      case 'clear':
+        for (const clearKey in clone) {
+          if (Object.prototype.hasOwnProperty.call(clone, clearKey)) {
+            delete clone[clearKey];
+          }
+        }
+
+        arr.push({ ...clone });
+
+        break;
+
+      case 'addProperties':
+        const data = actionObj.extraData;
+
+        for (const dataKey in data) {
+          const keyName = dataKey;
+          const keyValue = data[dataKey];
+
+          clone[keyName] = keyValue;
+        }
+
+        arr.push({ ...clone });
+
+        break;
+
+      case 'removeProperties':
+        const remove = actionObj.keysToRemove;
+
+        for (const stateKey in clone) {
```

Comment:
why do you do for loop by clone?

for (let key of remove) {
delete clone[key];
}